commands/migrate_commands.go

```diff
@@ -194,64 +253,77 @@ var gotoCommand = cli.Command{
 
 		log.Infof("Migrating to version %d", toVersionInt)
 
-		currentVersion, err := migrate.Version(ctx.GlobalString("url"), ctx.GlobalString("path"))
+		migrate, mctx, cancel := newMigrateWithCtx(ctx.GlobalString("url"), ctx.GlobalString("path"))
+		defer cancel()
+		currentVersion, err := migrate.Version(mctx)
 		if err != nil {
 			logErr(err).Fatalf("failed to migrate to version %d", toVersionInt)
 		}
 
 		relativeNInt := toVersionInt - int(currentVersion)
 
-		pipe := pipep.New()
-		go migrate.Migrate(pipe, ctx.GlobalString("url"), ctx.GlobalString("path"), relativeNInt)
-		ok := readPipe(pipe)
-		if !ok {
-			os.Exit(1)
+		err = migrate.Migrate(mctx, relativeNInt)
+		if err != nil {
+			logErr(err).Fatalf("Failed to migrate to vefrsion %d", toVersionInt)
 		}
-		logCurrentVersion(ctx.GlobalString("url"), ctx.GlobalString("path"))
+		logCurrentVersion(mctx, migrate)
 		return nil
 	},
 }
 
-func logErr(err error) *log.Entry {
-	return log.WithError(err)
+func newMigrateWithCtx(url, migrationsPath string) (*migrate.Handle, context.Context, func()) {
```

Comment:
So, why the `context.Context` is not a field of `*migrate.Handle`?
I smell a wrong utilization of context here.

Reply:
It shouldn't. If you add context to Handle at create time, it will make this Handle useless once you cancel this context.